src/serde_helpers.rs

```diff
@@ -437,16 +423,12 @@ pub mod uuid_as_java_legacy_binary {
         D: Deserializer<'de>,
     {
         let binary = Binary::deserialize(deserializer)?;
-        if binary.subtype != BinarySubtype::UuidOld {
-            Err(de::Error::custom("expecting BinarySubtype::UuidOld"))
-        } else if binary.bytes.len() != 16 {
-            Err(de::Error::custom("expecting 16 bytes"))
+        let res = binary.to_uuid_with_representation(UuidRepresentation::JavaLegacy);
```

Comment:
I'm not thrilled with this unpacking and then repacking of the result of `to_uuid` but returning the result directly doesn't type check.  I'm not sure if it's possible to constrain `D::Error` to be of type `crate::de::Error` but it doesn't compile when I directly attempt the constraint.